aws/defaults/handlers.go

```diff
@@ -189,12 +189,12 @@ var AfterRetryHandler = aws.NamedHandler{
 	Fn: func(r *aws.Request) {
 		// If one of the other handlers already set the retry state
 		// we don't want to override it based on the service's state
-		if r.Retryable == nil || r.Config.EnforceShouldRetryCheck {
-			r.Retryable = aws.Bool(r.ShouldRetry(r))
+		if r.Retryable == nil {
+			r.Retryable = aws.Bool(r.Retryer.ShouldRetry(r))
```

Comment:
I think the `Retryer.ShouldRetry` should be called regardless of the state of the `r.Retryable` flag. The SDK's `DefaultRetryer` should inspect this flag and if set, the DefaultRetryer should return that value, true|false.

This would remove the need for the `nil` check here in this function.

Reply:
Default Retryer already checks for r.Retryable flag in ShouldRetry fn. If it is set, we return the *r.Retryable.